permission-element.bs

```diff
@@ -19,7 +18,408 @@ Abstract: A `<permission>` HTML element to request browser permissions in-page.
+pepc = document.createElement("permission");
+pepc.type = "icecream";  // Ice cream is not a powerful browser feature. Not okay.
+pepc.type;  // "". Reflects the internal state.
+pepc.type = "camera";  // Still Not okay, because type as already been set.
+                       // Would have been okay as initial assignment.
+pepc.type;  // "". Reflects the internal state, which has not changed.
+
+```
+
+<div algorithm="HTMLPermissionElement/types getter">
+The HTMLPermissionElement's {{HTMLPermissionElement/type}} getter steps are:
+
+1. If {{[[Types]]}} is null: Return `""`.
+1. Return a string, containing the concatenation of all [=powerful feature=]
+    names in {{[[Types]]}}, seperated by " ".
+
+</div>
+
+<div algorithm="HTMLPermissionElement/types setter">
+The HTMLPermissionElement's {{HTMLPermissionElement/type}} setter steps are:
+
+1. If {{[[Types]]}} is not null: Return.
+1. Set {{[[Types]]}} to &laquo;[]&raquo;.
+1. Parse the input as a string of [=powerful feature=] names, seperated by whitespace.
+1. If any errors occured, return.
+1. Check if the set of [=powerful features=] is supported for the {{HTMLPermissionElement}} by the user agent. If not, return.
+1. Append each [=powerful feature=] name to the {{[[Types]]}} [=ordered set=].
+
+</div>
+
+### Activation blockers ### {#permission-element-activation-blockers}
+
+<pre class=idl>
+enum PermissionElementInvalidReason {
+  "",  // No blocker reason.
+  "style", "type_count", "illegal_subframe", "covered",
+  "recently_moved", "recently_created"
+};
+</pre>
+
+Issue: Is covered and visible the same type of blocker?
+
+Issue: Are unsupported types also a blocker?
+
+<div class="issue">
+In the Chrome implementation, I find these reasons:
+    * type_invalid
+    * illegal_subframe
+    * unsuccesful_registration
+    * recently_attached
+    * intersection_visible
+    * intersection_changed
+    * intersection_out_of_viewport_or_clipped
+    * intersection_occluded_or_distorted
+    * style_invalid
+
+</div>
+
+The permission element keeps track of "blockers", reasons why the element (currently) cannot be activated. These blockers come with three lifetimes: Permanent, temporary, and expiring.
+
+: Permanent blocker
+:: Once an element has a permanent blocker, it will be disabled permanently.
+    There are used for issues that the website owner is expected to fix.
+    An example is a <{permission}> element inside a <{fencedframe}>.
+: Temporary blocker
+:: This is a blocker that will only be valid until the blocking condition no
+    no longer occurs. An example is a <{permission}> element that is not
+    currently in view.
+: Expiring blocker
+:: This is a blocker that is only valid for a fixed period of time. This is
+    used to block abuse scenarios like "click jacking". An example is
+    a <{permission}> element that has recently been moved.
+
+<div>
+<dfn dfn lt="blocker reason table"></dfn>
+<table class="def">
+<tr><th>style
+<td>permanent (???)
```

Comment:
style should be "temporary" by the definitions above. If the style is fixed by the developer the element will become valid again (after a 500ms delay).

Reply:
Done.